rawhttp-core/src/main/java/rawhttp/core/body/encoding/GZipUncompressorOutputStream.java

```diff
@@ -69,23 +70,42 @@ public void flush() throws IOException {
 
     @Override
     public void finishDecoding() throws IOException {
-        super.finishDecoding();
-        encodedBytesSink.close();
-
         try {
+            super.finishDecoding();
+            closeQuietly(encodedBytesSink);
             readerExecution.get(5, TimeUnit.SECONDS);
         } catch (InterruptedException e) {
-            executorService.shutdownNow();
             throw new RuntimeException(e);
         } catch (ExecutionException e) {
-            executorService.shutdownNow();
-            throw new RuntimeException(e.getCause());
+            Throwable cause = e.getCause();
+            if (cause instanceof WrappedException) {
```

Comment:
I think that currently, this flow will not be reached. The WrappedException is only thrown at the reader thread.